mobilesdk/zbox/cache.go

```diff
@@ -0,0 +1,43 @@
+package zbox
+
+import (
+	"sync"
+	"time"
+
+	"github.com/0chain/gosdk/zboxcore/sdk"
+)
+
+type cachedAllocation struct {
+	Expiration time.Time
+	Allocation *sdk.Allocation
+}
+
+var (
+	cachedAllocations      = make(map[string]*cachedAllocation)
```

Comment:
Let's use `github.com/hashicorp/golang-lru/v2` for the cache to avoid the cache size increase without limitation. And it should be thread safe as well.

Reply:
fixed